odoo/addons/base/tests/test_res_company.py

```diff
@@ -50,3 +50,17 @@ def test_logo_check(self):
         self.assertTrue(company.uses_default_logo)
         company.partner_id.image_1920 = "R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
         self.assertFalse(company.uses_default_logo)
+
+    def test_unlink_company_with_children(self):
+        """Ensure that companies with child companies cannot be deleted."""
+
+        parent_company = self.env['res.company'].create({
+            'name': 'Parent Company',
+            'child_ids': [
+                Command.create({'name': 'Child Company'}),
+            ],
+        })
+
+        with self.assertRaisesRegex(UserError, 'Companies that have associated branches cannot be deleted'):
```

Comment:
Not need to check the content of the error. Moreover it will easier for the FW in master.